### PR TITLE
Refactor for extraction docs

### PR DESCRIPTION
**Pull request by fpingham**
Refactor for the extraction use case documentation

### CODE REVIEW

File: docs/extras/use_cases/extraction/index.mdx [hunk 1]
```diff
@@ -12,13 +14,106 @@ Examples of this include:
 - Extracting multiple rows to insert into a database from a long document
 - Extracting the correct API parameters from a user query
 
-This work is extremely related to [output parsing](/docs/modules/model_io/output_parsers/).
-Output parsers are responsible for instructing the LLM to respond in a specific format.
-In this case, the output parsers specify the format of the data you would like to extract from the document.
-Then, in addition to the output format instructions, the prompt should also contain the data you would like to extract information from.
+## Overview
+
+In LangChain we provide a few useful abstractions that help you leverage OpenAI [function calling](https://openai.com/blog/function-calling-and-other-api-updates) (`-0613` models) and which go a long way to avoid model hallucination. This is in general the best go-to to extract structured data from text using langchain. However there's other two options that you should also consider:
```
Comment (baskaryan) [inline docs/extras/use_cases/extraction/index.mdx:19]: i believe that the latest gpt-4 and gpt-3.5 models come with functions, don't need to be -0613 specifically anymore

**Review by rlancemartin — COMMENTED** [review]
(Updated into more granular comments below.)

File: docs/extras/use_cases/extraction/index.mdx (hunk 1 above)
Comment (rlancemartin) [reply docs/extras/use_cases/extraction/index.mdx:19]: also, now Anthropic has function so it seem we can broader the story beyond OAI.

File: docs/extras/use_cases/extraction/index.mdx [hunk 2]
```diff
@@ -4,6 +4,8 @@ sidebar_position: 2
 
 # Extraction
 
+## Use case
+
 Most APIs and databases still deal with structured information.
```
Comment (rlancemartin) [inline docs/extras/use_cases/extraction/index.mdx:9]: IMO the lede here should be more clear. E.g, something like - 

```
LLMs have incredible capacity for text generation.

But suppose you want the output formatted as JSON or in some other specified schema.

There are various approaches to to this, including (Pydantic, others?) output parsers and (OAI, Anthropic, others?) functions.
```

**Comment by rlancemartin** [comment]
One higher-level strategy point: @baskaryan do you think we should support [the option to spin up a Colab](https://www.endtoend.ai/blog/githubtocolab/) based on any use case doc? IMO would certainly be cool, but I don't have a sense for the cost. At least, it seems then all the use-case docs need to be Jupyter notebooks?

**Comment by rlancemartin** [comment]
> One higher-level strategy point: @baskaryan do you think we should support [the option to spin up a Colab](https://www.endtoend.ai/blog/githubtocolab/) based on any use case doc? IMO would certainly be cool, but I don't have a sense for the cost. At least, it seems then all the use-case docs need to be Jupyter notebooks?

Confirmed from discussion that we should do this.

But we can take up the work to convert to notebooks later.

File: docs/extras/use_cases/extraction/index.mdx [hunk 3]
```diff
@@ -4,6 +4,8 @@ sidebar_position: 2
 
 # Extraction
 
+## Use case
```
Comment (rlancemartin) [inline docs/extras/use_cases/extraction/index.mdx:7]: Maybe something more explicit like:

```
Getting structured output from LLM generation is hard.

For example, suppose you need the model output formatted as JSON or in some other specified schema.

Two primary approach have emerged for this:
* Functions
* Output parsing
```

File: docs/extras/use_cases/extraction/index.mdx (hunk 1 above)
Comment (rlancemartin) [inline docs/extras/use_cases/extraction/index.mdx:19]: Here we might offer a simple [schematic](https://docs.google.com/presentation/d/1zHk0Wwmvl_RM3FkiBZaOrts0cVoPHCcT0s6cmSwnSdo/edit?usp=sharing) to explain the differences:

image

File: docs/extras/use_cases/extraction/index.mdx [hunk 4]
```diff
@@ -12,13 +14,106 @@ Examples of this include:
...
+1. This work is extremely related to [output parsing](docs/modules/model_io/output_parsers/). Output parsers are responsible for instructing the LLM to respond in a specific format. We suggest using output parsers when either you are not using OpenAI models or you want to have very close control over the parsing (e.g. checking that the output is a date in a specific datetime format).
+2. While the conventional extraction chain is good enough for basic structuring of response data,
+   when doing extraction you often want to extract more complicated or nested structures.
+   For a deep dive on extraction, we recommend checking out [`kor`](https://eyurtsev.github.io/kor/),
+   a library that uses the existing LangChain chain and OutputParser abstractions
+   but deep dives on allowing extraction of more complicated schemas.
+
+All this said, let's see how easy it is to quickly and accurately extract structured data with LangChain.
+
+## Example #1: using a JSON schema
```
Comment (rlancemartin) [inline docs/extras/use_cases/extraction/index.mdx:30]: Maybe we re-title this to `Quickstart`: OAI function are probably the quickest way to get started.

File: docs/extras/use_cases/extraction/index.mdx [hunk 5]
```diff
@@ -12,13 +14,106 @@ Examples of this include:
...
+     Properties(person_name='Claudia', person_height=6, person_hair_color='brunette', dog_breed=None, dog_name=None)]
+
+## Going deeper
 
-While normal output parsers are good enough for basic structuring of response data,
-when doing extraction you often want to extract more complicated or nested structures.
-For a deep dive on extraction, we recommend checking out [`kor`](https://eyurtsev.github.io/kor/),
-a library that uses the existing LangChain chain and OutputParser abstractions
-but deep dives on allowing extraction of more complicated schemas.
+As we have seen, by leveraging OpenAI `-0613` models we can extract structured data from unstructured documents with minimal hallucination. For more detail on how to use these chains and different tips and tricks you can use, please check out the [docs](docs/extras/modules/chains/additional/extraction.ipynb) for the extraction chain.
```
Comment (rlancemartin) [inline docs/extras/use_cases/extraction/index.mdx:119]: Maybe we add a final section on `Parsing`

1/ Include the [JSON example](https://python.langchain.com/docs/modules/model_io/output_parsers/pydantic).

2/ Going Deeper

* Link to broader set of [parser docs](https://python.langchain.com/docs/modules/model_io/output_parsers/)

**Comment by baskaryan** [comment]
think we're missing an image `extraction_trace_function_2.png` @fpingham @rlancemartin

**Comment by rlancemartin** [comment]
> think we're missing an image `extraction_trace_function_2.png` @fpingham @rlancemartin

added! also scrubbed the ntbk; should build!

**Pull request merged**